### DISCUSSION TITLE
useFieldArray using Controller

### DISCUSSION THREAD

**Discussion by eden-lane** [react-hook-form/react-hook-form#1796] [Q&A] [answered]
Is it possible to use `useFieldArray` along with `Controller`?
When I log values into console I get
```
{
  events[1].mentor: "Sergey", 
  events[0].mentor: "Dmitry1"
}
```
instead of array with my values.

Here is [my codesandbox](https://codesandbox.io/s/blue-sound-fhos1?file=/src/App.js)

**Comment by eden-lane** [comment, marked as answer]
I saw an example for Field Arrays using Controller but I thought it's not what I am looking for because it logged same broken results:
```
{
  at: "", 
  fieldc2404ed0-a817-11ea-be06-3baf40d1ed0d: "c2404ed0-a817-11ea-be06-3baf40d1ed0d", 
  fieldc2b9eba0-a817-11ea-be06-3baf40d1ed0d: "c2b9eba0-a817-11ea-be06-3baf40d1ed0d"
}
```

actually the example itself is broken. 55's line should be
```
name={`field[${item.id}].value`}
```
instead of 
```
name={`field${item.id}`}
```

**Reply by talhabalaj** [reply]
This solved your issue?

**Reply by eden-lane** [reply]
No, I still get wrong results for some reason
https://codesandbox.io/s/cool-haslett-1i9wt?file=/src/App.js

**Reply by talhabalaj** [reply]
You have to pass `nest: true` to getValues function.
```javascript
getValues({ nest: true })
```

**Reply by eden-lane** [reply]
You're my saviour! Thanks!

**Reply by talhabalaj** [reply]
Pleasure is all mine

**Reply by bluebill1049** [reply]
thank you @talhabalaj FYI guys V6 will return nested by default.

**Reply by xolesteroll** [reply]
Mate i broke my head, while was getting it to work!!! Thank you so much!

**Comment by talhabalaj** [comment]
You forgot add a ref={register()}. Here's a working fork.
https://codesandbox.io/s/exciting-banzai-w8nrz

**Reply by eden-lane** [reply]
Unfortunately I can't use `register` because I have a custom component

**Comment by bluebill1049** [comment]
here is an example for controller: https://codesandbox.io/s/react-hook-form-usefieldarray-vy8fv
